CodingStyle.h

```diff
@@ -0,0 +1,93 @@
+/*=====================================================================
+ 
+ QGroundControl Open Source Ground Control Station
+ 
+ (c) 2009 - 2015 QGROUNDCONTROL PROJECT <http://www.qgroundcontrol.org>
+ 
+ This file is part of the QGROUNDCONTROL project
+ 
+ QGROUNDCONTROL is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+ 
+ QGROUNDCONTROL is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+ 
+ You should have received a copy of the GNU General Public License
+ along with QGROUNDCONTROL. If not, see <http://www.gnu.org/licenses/>.
+ 
+ ======================================================================*/
+
+// The above copyright block should be at the top of every file.
+
+/// @file
+///     @brief This is an example class header file which is used to describe the QGroundControl
+///             coding style. In general almost everything in here has some coding style meaning.
+///             For example, note how this doxygen block uses the /// variant as the preferred style
+///             of doxygen marker. Not all style choices are explained.
+///
+///     @author Don Gagne <don@thegagnes.com>
+
+#ifndef CodingStyle_h
```

Comment:
I'm matching the file naming convention here for this one #ifdef since I think it is more readable.